### PR TITLE
Features/#590 filter isolated gas buses

### PR DESCRIPTION
**Pull request by AmeliaNadal**
Fixes third point of https://github.com/openego/eGon-data/issues/590 .

## Before merging into `dev`-branch, please make sure that

- [x] the `CHANGELOG.rst` was updated.
- [x] new and adjusted code is formated using `black` and `isort`.
- [x] the `Dataset`-version is updated when existing datasets are adjusted.
- [x] the branch was merged into the
      `continuous-integration/run-everything-over-the-weekend`-branch.
- [x] the workflow is running successful in `test mode`.
- [x] the workflow is running successful in `Everything` mode.

### CODE REVIEW

File: src/egon/data/datasets/gas_grid.py [hunk 1]
```diff
@@ -660,6 +661,33 @@ def insert_gas_pipeline_list(
...
+    -------
+    None.
+    """
+    targets = config.datasets()["gas_grid"]["targets"]
+
+    db.execute_sql(
+        f"""
+        DELETE FROM {targets['buses']['schema']}.{targets['buses']['table']}
+        WHERE "carrier" = 'CH4'
+        AND scn_name = 'eGon2035'
```
Comment (CarlosEpia) [inline src/egon/data/datasets/gas_grid.py:677]: Is the problem about isolated gas buses happening only in the scenario eGon2035 or does it make sense for you to use this function also for the eGon100RE scenario?

**Review by CarlosEpia — COMMENTED** [review]
It looks good to me. Once we agree on extending the use of the new function or not to the eGon100RE, it can be merged.

File: src/egon/data/datasets/gas_grid.py (hunk 1 above)
Comment (AmeliaNadal) [reply src/egon/data/datasets/gas_grid.py:677]: The buses gas buses of eGon100RE are a copy a the gas buses of eGon2035 so we do not need to apply this function for the eGon100RE scenario.

Comment (CarlosEpia) [reply src/egon/data/datasets/gas_grid.py:677]: Thanks for the clarification.

**Review by CarlosEpia — APPROVED** [review]
This PR can be merged.

**Pull request merged**